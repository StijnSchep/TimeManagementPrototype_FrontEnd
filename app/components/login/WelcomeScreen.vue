<template>
    <Page class="body" actionBarHidden="true">
        <FlexboxLayout class="page">
            <Label :text="'Welkom ' + employee.firstname + ' ' + employee.lastname" class="welcome-txt"/>
            <ActivityIndicator :busy="busy" color="#fff" @busyChange="onBusyChanged" />
        </FlexboxLayout>
    </Page>
</template>

<script>
    import Dashboard from "../dashboard/Dashboard";

    export default {
        props: ['employee', 'token'],
        data() {
            return {
                busy: true
            }
        },
        methods: {
            onBusyChanged() {
                this.goToDashboard()
            },
            goToDashboard() {
                this.$navigateTo(Dashboard, {
                    clearHistory: true,
                    transition: {
                        name: 'fade',
                        duration: 400
                    },
                    props: {
                        token: this.token
                    }
                })
            },
            sleep(self) {
                setTimeout(() => {
                    self.busy = false;
                }, 2000)
            }
        },
        created() {
            this.sleep(this);
        }
    }
</script>

<style scoped>
    .body {
        background-image: url("~/assets/images/SplashScreen.jpg");
        background-repeat: no-repeat;
        background-position: center top;
        background-size: cover;
    }

    .page {
        align-items: center;
        vertical-align: center;
        flex-direction: column;
    }

    .welcome-txt {
        text-align: center;
        color: white;
        font-size: 30px;
        margin-bottom: 10px;
    }
</style>
