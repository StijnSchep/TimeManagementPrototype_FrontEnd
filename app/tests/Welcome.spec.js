import { shallowMount, mount } from '@vue/test-utils';
import WelcomeScreen from '../components/login/WelcomeScreen';


jest.mock('tns-core-modules/application-settings', () => ({
  getString: jest.fn(() => "")
}));

jest.mock('nativescript-fingerprint-auth', () => ({
  FingerprintAuth: jest.fn(),
  getString: jest.fn(() => "")
}));

jest.mock('tns-core-modules/application', () => ({
  android: {
    on: jest.fn()
  },
  AndroidApplication: {
    activityBackPressedEvent: {}
  }
}));

jest.mock("tns-core-modules/ui/dialogs", () => ({
  alert: jest.fn(() => Promise.resolve())
}));


const wrapper = shallowMount(WelcomeScreen, {
  propsData: {
    employee: Object,
    token: String
  }
});

describe('WelcomeScreen.vue', () => {
  it('mounts and renders', async () => {
    await expect(wrapper.html()).toBeTruthy();
  });

  test('renders correctly', async () => {
    await expect(wrapper.element).toMatchSnapshot();
  });

  it('is a Vue instance', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('Expect component to exist', async () => {
    await expect(wrapper).toBeTruthy();
  });

  it('Expect component to have Page element', async () => {
    await expect(wrapper.find('Page')).toBeTruthy();
  });

  it('Expect component to have GridLayout element', async () => {
    await expect(wrapper.find('GridLayout')).toBeTruthy();
  });

  it('Expect component to have Label element', async () => {
    await expect(wrapper.find('Label')).toBeTruthy();
  });

  it('Expect component to have data property', async () => {
    await expect(typeof WelcomeScreen.data).toBe('function')
    const defaultData = WelcomeScreen.data()
    await expect(defaultData.busy).toBe(true)
  });

  it('Expect component to have methods property which is an object', async () => {
    await expect(typeof WelcomeScreen.methods).toBe('object');
  });

  it('Expect methods object to have onBusyChanged', async () => {
    await expect(typeof WelcomeScreen.methods.onBusyChanged).toBe('function');
  });

  it('Expect methods object to have goToDashboard function', async () => {
    await expect(typeof WelcomeScreen.methods.goToDashboard).toBe('function');
  });

  it('Expect methods object to have sleep fucntion', async () => {
    await expect(typeof WelcomeScreen.methods.sleep).toBe('function');
  });

  it('Expect created() lifecylce hook', async () => {
    await expect(typeof WelcomeScreen.created).toBe('function');
  });
});


