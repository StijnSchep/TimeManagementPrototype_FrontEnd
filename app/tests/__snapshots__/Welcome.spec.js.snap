// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen.vue renders correctly 1`] = `
<page-stub
  actionbarhidden="true"
  class="body"
>
  <flexboxlayout-stub
    class="page"
  >
    <label
      class="welcome-txt"
      text="Welkom undefined undefined"
    />
     
    <activityindicator-stub
      busy="true"
      color="#fff"
    />
  </flexboxlayout-stub>
</page-stub>
`;
